v2/connection.go

```diff
@@ -71,76 +52,33 @@ func connectionHandler(c *DeviceClient) {
 				}
 			}
 
-		case <-c.stableTimerCh:
-			c.dbg.print("Stable timer reached\n")
-			c.stateUpdateCh <- stable
-
 		case state = <-c.stateUpdateCh:
 			c.dbg.printf("State updated to %s\n", state.String())
 
 			switch state {
 			case inactive:
-				c.stableTimerCh = make(chan bool)
-				c.reconnectPeriod = c.reconnectPeriod * 2
-				if c.reconnectPeriod > c.opt.MaximumReconnectTime {
-					c.reconnectPeriod = c.opt.MaximumReconnectTime
-				}
-				c.dbg.printf("Trying to reconnect (%d ms)\n", c.reconnectPeriod/time.Millisecond)
-				go func() {
-					time.Sleep(c.reconnectPeriod)
-					c.stateUpdateCh <- reconnecting
-				}()
-
-			case reconnecting:
-				c.connect()
-				psq.resubscribe()
-
+				// do nothing
 			case established:
 				c.dbg.print("Processing queued operations\n")
-				go func() {
-					time.Sleep(c.opt.MinimumConnectionTime)
-					c.stableTimerCh <- true
-				}()
 				psq.processQueuedOps()
-
-			case stable:
-				if statePrev == established {
-					c.dbg.print("Connection is stable\n")
-					c.reconnectPeriod = c.opt.BaseReconnectTime
-				}
-
 			case terminating:
 				c.dbg.print("Terminating connection\n")
 				c.cli.Disconnect(250)
 				return
-
 			default:
 				panic("Invalid internal state\n")
 			}
 		}
 	}
 }
 
-func (s *pubSubQueues) publishOrEnqueue(d *pubqueue.Data) {
-	token := s.cli.cli.Publish(d.Topic, s.cli.opt.Qos, s.cli.opt.Retain, d.Payload)
-	go func() {
-		token.Wait()
-		if token.Error() != nil {
```

Comment:
maybe, this error handling is still required for https://github.com/seqsense/aws-iot-device-sdk-go/pull/39#issuecomment-521516827